clients/src/main/java/org/apache/kafka/clients/consumer/internals/CompletedFetch.java

```diff
@@ -311,25 +312,33 @@ <K, V> ConsumerRecord<K, V> parseRecord(Deserializers<K, V> deserializers,
                                             Optional<Integer> leaderEpoch,
                                             TimestampType timestampType,
                                             Record record) {
+        long offset = record.offset();
+        long timestamp = record.timestamp();
+        ByteBuffer keyBytes = record.key();
+        ByteBuffer valueBytes = record.value();
+        Headers headers = new RecordHeaders(record.headers());
+        K key;
+        V value;
         try {
-            long offset = record.offset();
-            long timestamp = record.timestamp();
-            Headers headers = new RecordHeaders(record.headers());
-            ByteBuffer keyBytes = record.key();
-            K key = keyBytes == null ? null : deserializers.keyDeserializer.deserialize(partition.topic(), headers, keyBytes);
-            ByteBuffer valueBytes = record.value();
-            V value = valueBytes == null ? null : deserializers.valueDeserializer.deserialize(partition.topic(), headers, valueBytes);
-            return new ConsumerRecord<>(partition.topic(), partition.partition(), offset,
-                    timestamp, timestampType,
-                    keyBytes == null ? ConsumerRecord.NULL_SIZE : keyBytes.remaining(),
-                    valueBytes == null ? ConsumerRecord.NULL_SIZE : valueBytes.remaining(),
-                    key, value, headers, leaderEpoch);
+            key = keyBytes == null ? null : deserializers.keyDeserializer.deserialize(partition.topic(), headers, keyBytes);
         } catch (RuntimeException e) {
-            log.error("Deserializers with error: {}", deserializers);
-            throw new RecordDeserializationException(partition, record.offset(),
-                    "Error deserializing key/value for partition " + partition +
+            throw new KeyDeserializationException(partition, offset, keyBytes, valueBytes, headers, record.timestamp(),
```

Comment:
I think you need a TimestampType also on the constructor for the `RecordDeserializationException`.